docs: local file dependencies for Checkly CLI (#690)
checkly/checklyhq.com: site/content/docs/cli/npm-packages.md

---
title: Using NPM packages and local dependencies
weight: 4
menu:
  docs:
    parent: "CLI"
---

Checkly lets you use JavaScript / TypeScript in your Browser checks and in API check setup & teardown scripts.
Checks are able to use NPM packages that are defined in [our runtimes](https://www.checklyhq.com/docs/runtimes/specs/), as well as import local JavaScript and TypeScript files.

## NPM packages

> Not all NPM packages from NPM are available inside the context of a Check.

The JavaScript code for checks executes in a runtime environment managed by Checkly.

> A runtime consists of a `runtimeId` which you can set at `Project` level or individual `Check` level.

Reviewer: @clample @tnolet via the UI I believe Runtimes can be set at Group level too. Is there a discrepancy with the CLI or is this incorrect?

Author: @ragog you're right, this can be set at the group level as well.

https://github.com/checkly/checkly-cli/blob/c4c97b2417e630106c878913cec5c4985a93bc33/packages/cli/src/constructs/check-group.ts#L50-L53

I'll update this to:

> Runtime versions can be selected by setting a `runtimeId`. This can be configured using the `Check` and `CheckGroup` constructs, and a default value for the project can be set in the [CLI global configuration](https://www.checklyhq.com/docs/cli/project-structure/#global-configuration).

- [Browse the latest runtime specs](https://www.checklyhq.com/docs/runtimes/specs/)
- [Learn more about runtimes](https://www.checklyhq.com/docs/runtimes/)

## Local Dependencies

> Browser Checks are also able to import other JavaScript and TypeScript files as dependencies.

Reviewer: @clample we are using "Browser check" with only the first word capitalized elsewhere in the page and the docs. Do we want to keep this consistent?

Reviewer: I do see this also appears as "Browser Checks" in other places. Not sure if the guidelines have changed...

Author: @ragog I can update the page to follow best practices. It should be "Browser Checks" everywhere, right?

Reviewer: @clample it turns out it is "browser checks" (unless in a headline or at the beginning of a phrase ofc) - see the new Checkly Style Guide on notion.

> This can be particularly useful for defining helper functions to be reused across multiple checks.

Reviewer 2: ```suggestion
This is useful for defining helper functions to be reused across multiple checks.
```

Let's save some words. "Can" is a word that can often be dropped to save words. Shorter and bolder is easier to read. :)

Author: Thanks, good idea

The Checkly CLI will automatically detect these dependencies and make sure that they're bundled as part of the check.
No additonal configuration is needed, and there's no need to create [Snippets](/docs/snippets/) in your account.

Here is an example of how this works in practice. The directory tree looks like the following:
```
__checks__
├── login.check.ts
├── login.spec.ts
└── login-helper.ts
```

In `login-helper.ts` we define a reusable function `gitHubLogin`. This function can be used by multiple Browser checks.

> ```

Reviewer: @clample I think we want to use syntax highlighting here and in most other code blocks

```
// @ts-ignore
import { test } from '@playwright/test'
import { gitHubLogin } from './login-helper'

test('Github login', async ({ page }) => {
  await gitHubLogin(page, process.env.GITHUB_USER, process.env.GITHUB_PWD)

  // your normal check code
  await page.click('.header-search-input')
})
```

Finally, in `login.check.ts` we define the [Browser Check construct](/docs/cli/constructs/#browsercheck). Note that it's only necessary to configure the main Playwright file `login.spec.ts`. The dependency on `login-helper.ts` is automatically detected by the CLI.
```
import { BrowserCheck } from '@checkly/cli/constructs'

new BrowserCheck('login-check', {
  name: 'Login Check',
    code: { entrypoint: './login.spec.ts' }
  })
})
```

After running [`npx checkly deploy`](/docs/cli/command-line-reference/#npx-checkly-deploy) to deploy the project, we can see in the Web UI that the helper file `login-helper.ts` was also uploaded for the check.

> ![login check with helper file in dependencies](/docs/images/cli/github_login_helper_dependency.png)

Reviewer: @clample take this as an optional suggestion, but I think the screenshot could use either a zoom in to the dependency bit, or a callout of some sort highlighting it

Reviewer 2: Agreed as optional, but a leading focus would be 💯 . Example (https://checklyhq.slack.com/archives/C03H435FU5R/p1677098383180029?thread_ts=1677095180.409809&cid=C03H435FU5R)
